pkg/controller/migmigration/migmigration_controller.go

```diff
@@ -318,3 +325,22 @@ func (r *ReconcileMigMigration) setOwnerReference(migration *migapi.MigMigration
 
 	return nil
 }
+
+// Ensures that the labels required to assist debugging are present on migmigration
+func (r *ReconcileMigMigration) ensureDebugLabels(migration *migapi.MigMigration) error {
+	if migration.Labels == nil {
+		migration.Labels = make(map[string]string)
+	} else {
+		if value, exists := migration.Labels[MigPlanDebugLabel]; exists {
+			if value == migration.Spec.MigPlanRef.Name {
+				return nil
+			}
+		}
+	}
+	migration.Labels[MigPlanDebugLabel] = migration.Spec.MigPlanRef.Name
```

Comment:
May want to do a nil check on `MigPlanRef` at the start of this method, as the `migPlanRef` field can be nil due to being a ptr with `omitempty` set.

https://github.com/konveyor/mig-controller/blob/90e9e8d9b983a149b4c2b4cb14ce75fc2c3c5aa0/pkg/apis/migration/v1alpha1/migmigration_types.go#L32